components/metrics/labels.go

```diff
@@ -46,3 +46,30 @@ func labelsFromCtx(ctx context.Context, labels ...string) prometheus.Labels {
 
 	return ctxLabels
 }
+
+type LabelComputeFn func(msgCtx context.Context) string
```

Comment:
We might want to reflect on that signature. Is it enough to use only the context to get a label's value? Should we also pass the message pointer? For my use case, the additional labels are static, so I basically do this
```
        metrics.MetricLabel{
		Label: "service",
		ComputeFn: func(ctx context.Context) string {
			return "my_service"
		},
	},
```
but there might be a use case for getting something from the message itself as a label value.